### DISCUSSION TITLE
Log margin means Emacs is unwilling to split horizontally

### DISCUSSION THREAD

**Discussion by spwhitton** [magit/magit#4681] [Ideas]
Hello,

While the log margin is shown, its width is subtracted from `(window-width)`.  In my case, this makes `(window-width)` lower than twice `split-window-threshold`, and so if I hit SPC on a commit I get a vertical split rather than a horizontal split.  This means that whether or not you get a horizontal split depends upon whether or not the log margin is shown, which I find unintuitive.  In particular, a horizontal split is preferable for looking through a commit's diff alongside viewing the log, and my frame is definitely big enough.

To reproduce just make your Emacs frame small enough that `(frame-width)` is approx. 170.  Hit SPC on a commit in the log buffer.  Now `C-x 1` and `L L` and hit SPC on a commit again.

Perhaps `magit-display-buffer` should hide the log and redisplay it afterwards if necessary, so `display-buffer` sees the true window width?

Thanks.

**Comment by tarsius** [comment]
> While the log margin is shown, its width is subtracted from `(window-width)`. In my case, this makes `(window-width)` lower than twice `split-window-threshold`, and so if I hit SPC on a commit I get a vertical split rather than a horizontal split. This means that whether or not you get a horizontal split depends upon whether or not the log margin is shown, which I find unintuitive. 

One could also argue that it behaves exactly as intended. The margin takes space away from the windows, causing the available width to cross the threshold specified by `split-height-threshold` to be crossed. With some experimentation, you might be able to change the value of that variable to prevent that while not changing the result in too many other situations.

> Perhaps `magit-display-buffer` should hide the log and redisplay it afterwards if necessary, so `display-buffer` sees the true window width?

I don't think that's something Magit should do. But this function already calls two hooks, one before everything else and one after everything else, so you should be able to implement this yourself.

**Comment by spwhitton** [comment]
Hello,

On Fri 06 May 2022 at 04:59AM -07, Jonas Bernoulli wrote:

>> While the log margin is shown, its width is subtracted from `(window-width)`. In my case, this makes `(window-width)` lower than twice `split-window-threshold`, and so if I hit SPC on a commit I get a vertical split rather than a horizontal split. This means that whether or not you get a horizontal split depends upon whether or not the log margin is shown, which I find unintuitive.
>
> One could also argue that it behaves exactly as intended. The margin takes space away from the windows, causing the available width to cross the threshold specified by `split-height-threshold` to be crossed. With some experimentation, you might be able to change the value of that variable to prevent that while not changing the result in too many other situations.
>
>> Perhaps `magit-display-buffer` should hide the log and redisplay it afterwards if necessary, so `display-buffer` sees the true window width?
>
> I don't think that's something Magit should do. But this function already calls two hooks, one before anything else and one after anything else, so you should be able to implement this yourself.

Fair enough, thanks.

-- 
Sean Whitton
